<a href="https://colab.research.google.com/github/sahilsagarprasad97-spec/flyrank-ml-week1/blob/main/Copy_of_w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/sahilsagarprasad97-spec/flyrank-ml-week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [8]:
!git clone https://github.com/sahilsagarprasad97-spec/flyrank-ml-week1.git

fatal: destination path 'flyrank-ml-week1' already exists and is not an empty directory.


In [9]:
import pandas as pd

df = pd.read_csv("/content/flyrank-ml-week1/data/raw/content_refresh_anonymized.csv")

print("Rows and Columns:", df.shape)

df.head()

Rows and Columns: (30000, 44)


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


## 1. My lane (or freestyle) and why

I have selected the **Refresh / Content Opportunity Scoring** lane.

I selected this lane because it focuses on identifying content that should be refreshed or improved based on available data. This can help prioritize pages that have the greatest opportunity for better search performance. It is a practical starting point for learning how machine learning can support content decision-making.

In [10]:
import pandas as pd

df = pd.read_csv("/content/flyrank-ml-week1/data/raw/content_refresh_anonymized.csv")

print("Rows and Columns:", df.shape)

df.head()

Rows and Columns: (30000, 44)


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


## 2. The question: decision, action, cost of a wrong call

**Research Question:**
Which content pages should be refreshed first to improve search performance?

**Decision:**
Identify and prioritize content pages that have the highest opportunity for improvement.

**Action:**
The SEO or content team will review, update, and optimize the selected pages.

**Cost of a Wrong Recommendation:**
If a low-priority page is selected, time and resources may be wasted. If a high-priority page is ignored, the business may miss valuable search traffic and potential customers.

In [11]:
print("Dataset Shape:", df.shape)

print("\nNumber of unique clients:")
print(df["client_id"].nunique())

print("\nAverage search volume:")
print(df["search_volume"].mean())

Dataset Shape: (30000, 44)

Number of unique clients:
32

Average search volume:
158.88239139909922


## 3. Quick look at the data

The starter dataset contains **30,000 rows** and **44 columns**, representing content and SEO-related information.

The dataset includes **32 unique clients**, allowing comparisons across multiple websites.

The average search volume is **158.88**, indicating that the dataset contains keywords with different levels of search demand.

These values show that the dataset is large enough to support meaningful analysis for identifying content refresh opportunities.

In [12]:
print("Dataset Shape:", df.shape)
print("Unique Clients:", df["client_id"].nunique())
print("Average Search Volume:", df["search_volume"].mean())

Dataset Shape: (30000, 44)
Unique Clients: 32
Average Search Volume: 158.88239139909922


## 4. Careful words: what I can and can't claim

This project can identify content pages that appear to have higher refresh opportunities based on patterns in the available data. It provides decision support for the content and SEO team.

However, this project cannot prove that refreshing a page will definitely improve Google rankings or traffic. The results should be treated as recommendations that require human review before taking action.

In [13]:
print("Project Type: Decision Support")

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nImportant Note:")
print("This analysis provides recommendations based on observed data patterns.")
print("It does not prove that refreshing content will guarantee better rankings.")

Project Type: Decision Support
Rows: 30000
Columns: 44

Important Note:
This analysis provides recommendations based on observed data patterns.
It does not prove that refreshing content will guarantee better rankings.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.